src/hooks/usePDFPreview.ts

```diff
@@ -67,59 +66,58 @@ export default function usePDFPreview(physicalDocument: TDocumentPage, documentP
 
   const annotationConfig = { showToolbar: false, showCommentsPanel: false, downloadWithAnnotations: true, printWithAnnotations: true };
 
-  let viewSDKClient = null;
-  let embedApi = null;
-  let annotationManagerApi = null;
+  let viewSDKClient_CACHE = null;
+  let embedApi_CACHE = null;
 
-  const createPDFClient = (startingPassage: number, onLoadCallback?: Function) => {
-    viewSDKClient = new ViewSDKClient();
-    viewSDKClient.ready().then(() => {
-      // Prevent attaching preview to non-existent div
-      const pdfDiv = document.getElementById("pdf-div");
-      if (!pdfDiv) return;
-      const previewFilePromise = viewSDKClient.previewFile(physicalDocument, adobeKey, "pdf-div", viewerConfig);
-      previewFilePromise.then((adobeViewer: any) => {
-        // PDF PREVIEW -- SHOULD BE VISIBLE NOW
-        adobeViewer.getAPIs().then((api: any) => {
-          embedApi = api;
-          // if we have a passage index, scroll to it
-          if (!isNaN(Number(startingPassage))) {
-            passageIndexChangeHandler(startingPassage);
-          }
-        });
-        adobeViewer.getAnnotationManager().then((annotationManager: any) => {
-          annotationManager.setConfig(annotationConfig);
-          annotationManagerApi = annotationManager;
-          if (documentPassageMatches.length > 0) {
-            // annotationManager.addAnnotations(generateHighlights(document, documentPassageMatches));
-            documentMatchesChangeHandler(documentPassageMatches);
-          }
-        });
-        if (onLoadCallback) {
-          onLoadCallback();
-        }
-      });
-    });
+  const createPDFClient = async () => {
```

Comment:
suggestion: would a name like `getAdobeApis` be more true to the return type?

Reply:
Critically, it also calls the `previewFile` promise method, which initialises the PDF into the HTML container on the page. The Adobe Apis can only then be accessed from the return of that promise.

In general I agree some of the names in this file could be more descriptive though, I'll review.

Reply:
> it also calls the previewFile promise method, which initialises the PDF into the HTML container on the page. The Adobe Apis can only then be accessed from the return of that promise.

This feels like exactly what a method called `getAdobeApis` should encapsulate and obfuscate from anyone using this API.

Reply:
Where this gets interesting and a bit trickier, is that I want to have access to the APIs but without re-running the `previewFile` method, because in the browser this is what triggers the document loader to re-load from scratch.

I feel like reloading is acceptable UX for the highlighting (mainly because of the out-of-memory issue), but for the page change (which relies upon the `embedApi` this is not acceptable).

Reply:
Don't disagree - ideally we wouldn't have this memory issue on the highlighter and then we could memoize the APIs